manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/BridgeWorker.java

```diff
@@ -210,15 +212,15 @@ protected boolean isDeprovisioningComplete(Bridge managedResource) {
     // The ErrorHandler Processor is a special type. It is not visible to Users and hence
     // should it fail to be provisioned (or de-provisioned) the User is unaware of the reason.
     // Therefore, propagate the error associated with the Error Handler Processor to the Bridge.
-    protected void propagateProcessorError(Bridge bridge) {
-        String bridgeId = bridge.getId();
-        String customerId = bridge.getCustomerId();
-        processorService.getErrorHandler(bridgeId, customerId)
-                .ifPresent(errorHandler -> {
-                    bridge.setErrorId(errorHandler.getErrorId());
-                    bridge.setErrorUUID(errorHandler.getErrorUUID());
-                    persist(bridge);
-                });
+    protected Bridge persistAndPropagateErrorHandlerFailure(Bridge bridge, Processor errorHandler) {
```

Comment:
Again, not sure how these two parts are related

Reply:
It's an internal implementation detail.

I need to persist the `Bridge` object before invoking `BridgeSerrvice.updateBridgeStatus(..)` since this loads the `Bridge` based on ID. I could refactor things to use an in-memory object but the `BridgeService` is a public service and I'd prefer not to pollute its interface to avoid this method.